### ISSUE TITLE
Android Dockerfile uses dev branch instead of local files

### ISSUE THREAD

**Issue by skyl** [zingolabs/zingo-mobile#86]
https://github.com/zingolabs/zingo-mobile/blob/dev/rust/android/Dockerfile#L94-L98

Is there some reason I'm missing to build `dev` instead of local files? Something like this

```
RUN cd /opt && git clone --depth=1 --branch="dev" https://github.com/zingolabs/zingo-mobile zingo
RUN printf '[net] \ngit-fetch-with-cli = true\n' > /root/.cargo/config.toml
RUN cargo fetch --manifest-path /opt/zingo/rust/android/Cargo.toml
RUN cargo build --release --manifest-path /opt/zingo/rust/android/Cargo.toml
RUN rm -rf /opt/zingo
```

should probably be changed to use the local files like:

```
COPY android/ /opt/zingo/rust/android/
COPY lib/ /opt/zingo/rust/lib/
RUN cargo fetch --manifest-path /opt/zingo/rust/android/Cargo.toml
RUN cargo build --release --manifest-path /opt/zingo/rust/android/Cargo.toml
RUN rm -rf /opt/zingo
```

A first attempt at this though somehow resulted in

```
#23 311.4 error[E0277]: the trait bound `Action<_>: ShieldedOutput<_, 580_usize>` is not satisfied
#23 311.4    --> /root/.cargo/git/checkouts/zingolib-2003bc87acd6fa69/ddf8261/lib/src/blaze/fetch_full_transaction.rs:570:9
#23 311.4     |
#23 311.4 570 |         scan_bundle(
#23 311.4     |         ^^^^^^^^^^^ the trait `ShieldedOutput<_, 580_usize>` is not implemented for `Action<_>`
#23 311.4     |
#23 311.4     = help: the following other types implement trait `ShieldedOutput<D, CIPHERTEXT_SIZE>`:
#23 311.4               <CompactOutputDescription as ShieldedOutput<SaplingDomain<P>, {_: usize}>>
#23 311.4               <OutputDescription<A> as ShieldedOutput<SaplingDomain<P>, {_: usize}>>
#23 311.4               <Unspendable as ShieldedOutput<SaplingDomain<zcash_primitives::consensus::MainNetwork>, ENC_CIPHERTEXT_SIZE>>
#23 311.4 note: required by a bound in `scan_bundle`
#23 311.4    --> /root/.cargo/git/checkouts/zingolib-2003bc87acd6fa69/ddf8261/lib/src/blaze/fetch_full_transaction.rs:611:8
#23 311.4     |
#23 311.4 591 | async fn scan_bundle<K, B, L, O, D, E>(
#23 311.4     |          ----------- required by a bound in this
#23 311.4 ...
#23 311.4 611 |     O: ShieldedOutput<D, ENC_CIPHERTEXT_SIZE>,
#23 311.4     |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `scan_bundle`
#23 311.4 
#23 311.4 error[E0277]: the trait bound `OrchardDomain: Domain` is not satisfied
#23 311.4    --> /root/.cargo/git/checkouts/zingolib-2003bc87acd6fa69/ddf8261/lib/src/blaze/fetch_full_transaction.rs:570:9
#23 311.4     |
#23 311.4 570 |         scan_bundle(
#23 311.4     |         ^^^^^^^^^^^ the trait `Domain` is not implemented for `OrchardDomain`
#23 311.4     |
#23 311.4     = help: the trait `Domain` is implemented for `SaplingDomain<P>`
#23 311.4 note: required by a bound in `scan_bundle`
#23 311.4    --> /root/.cargo/git/checkouts/zingolib-2003bc87acd6fa69/ddf8261/lib/src/blaze/fetch_full_transaction.rs:610:8
#23 311.4     |
#23 311.4 591 | async fn scan_bundle<K, B, L, O, D, E>(
#23 311.4     |          ----------- required by a bound in this
#23 311.4 ...
#23 311.4 610 |     D: Domain,
#23 311.4     |        ^^^^^^ required by this bound in `scan_bundle`
#23 311.4 
#23 312.1 error[E0277]: the trait bound `OrchardDomain: Domain` is not satisfied
#23 312.1    --> /root/.cargo/git/checkouts/zingolib-2003bc87acd6fa69/ddf8261/lib/src/blaze/trial_decryptions.rs:234:33
#23 312.1     |
#23 312.1 233 | ...                   zcash_note_encryption::try_compact_note_decryption(
#23 312.1     |                       -------------------------------------------------- required by a bound introduced by this call
#23 312.1 234 | ...                       &OrchardDomain::for_nullifier(action.nullifier()),
#23 312.1     |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Domain` is not implemented for `OrchardDomain`
#23 312.1     |
#23 312.1     = help: the trait `Domain` is implemented for `SaplingDomain<P>`
#23 312.1 note: required by a bound in `zcash_note_encryption::try_compact_note_decryption`
#23 312.1    --> /root/.cargo/git/checkouts/librustzcash-0a74bd38a00f78b0/8ad60a0/components/zcash_note_encryption/src/lib.rs:609:39
#23 312.1     |
#23 312.1 609 | pub fn try_compact_note_decryption<D: Domain, Output: ShieldedOutput<D, COMPACT_NOTE_SIZE>>(
#23 312.1     |                                       ^^^^^^ required by this bound in `zcash_note_encryption::try_compact_note_decryption`
#23 312.1 
#23 312.1 error[E0277]: the trait bound `CompactAction: ShieldedOutput<OrchardDomain, 52_usize>` is not satisfied
#23 312.1    --> /root/.cargo/git/checkouts/zingolib-2003bc87acd6fa69/ddf8261/lib/src/blaze/trial_decryptions.rs:236:33
#23 312.1     |
#23 312.1 233 | ...                   zcash_note_encryption::try_compact_note_decryption(
#23 312.1     |                       -------------------------------------------------- required by a bound introduced by this call
#23 312.1 ...
#23 312.1 236 | ...                       &action,
#23 312.1     |                           ^^^^^^^ the trait `ShieldedOutput<OrchardDomain, 52_usize>` is not implemented for `CompactAction`
#23 312.1     |
#23 312.1     = help: the following other types implement trait `ShieldedOutput<D, CIPHERTEXT_SIZE>`:
#23 312.1               <CompactOutputDescription as ShieldedOutput<SaplingDomain<P>, {_: usize}>>
#23 312.1               <OutputDescription<A> as ShieldedOutput<SaplingDomain<P>, {_: usize}>>
#23 312.1               <Unspendable as ShieldedOutput<SaplingDomain<zcash_primitives::consensus::MainNetwork>, ENC_CIPHERTEXT_SIZE>>
#23 312.1 note: required by a bound in `zcash_note_encryption::try_compact_note_decryption`
#23 312.1    --> /root/.cargo/git/checkouts/librustzcash-0a74bd38a00f78b0/8ad60a0/components/zcash_note_encryption/src/lib.rs:609:55
#23 312.1     |
#23 312.1 609 | pub fn try_compact_note_decryption<D: Domain, Output: ShieldedOutput<D, COMPACT_NOTE_SIZE>>(
#23 312.1     |                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `zcash_note_encryption::try_compact_note_decryption`
#23 312.1 
#23 312.1 error[E0277]: the trait bound `OrchardDomain: Domain` is not satisfied
#23 312.1    --> /root/.cargo/git/checkouts/zingolib-2003bc87acd6fa69/ddf8261/lib/src/blaze/trial_decryptions.rs:235:33
#23 312.1     |
#23 312.1 235 | ...                   &ivk,
#23 312.1     |                       ^^^^ the trait `Domain` is not implemented for `OrchardDomain`
#23 312.1     |
#23 312.1     = help: the trait `Domain` is implemented for `SaplingDomain<P>`
#23 312.1 
#23 322.0 For more information about this error, try `rustc --explain E0277`.
#23 322.2 error: could not compile `zingolib` due to 5 previous errors
```

In #77 at 86d3a76df16f0ca67ae3cb0c3795b3d4c568cc5d

**Comment by skyl** [comment]
for https://github.com/zingolabs/zingo-mobile/pull/77 the problem was solved by adding the workspace Cargo.lock. But, this indicates breakage on the way ... this issue can either be closed or used to track the incoming breakage ...

**Comment by AloeareV** [comment]
Ah! I'm familiar with this error, we're using patched versions of dependencies in zingolib. According to https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#working-with-an-unpublished-minor-version, patching applies transitively at the top level, which means we need to add these lines to our `rust/Cargo.toml` in order to have the patch apply here: https://github.com/zingolabs/zingolib/blob/ddf8261a2a6aabb5a1fb8b46d450d7ed5aee339c/Cargo.toml#L11-L14

I hadn't realized that when originally patching, this is an additional hassle for any crates that consume zingolib, so I'm hoping we can find an alternate solution in the near future, because keeping the downstream patched versions aligned with the zingolib patched versions sounds like absolute hell.

**Comment by skyl** [comment]
can open a new issue if the stacktrace from the OP comes back. But, the dockerfile android build now uses the local filesystem. so, this issue can be closed by #77
